packages/service-library/src/servicelib/aiohttp/profiler_middleware.py

```diff
@@ -0,0 +1,41 @@
+from aiohttp.web import HTTPInternalServerError, Request, StreamResponse, middleware
+from pyinstrument import Profiler
+from servicelib.mimetype_constants import (
+    MIMETYPE_APPLICATION_JSON,
+    MIMETYPE_APPLICATION_ND_JSON,
+)
+
+from .._utils_profiling_middleware import append_profile
+
+
+@middleware
+async def profiling_middleware(request: Request, handler):
+    profiler: Profiler | None = None
+    if request.headers.get("x-profile") is not None:
+        profiler = Profiler(async_mode="enabled")
+        profiler.start()
+
+    response = await handler(request)
+
+    if profiler is None:
+        return response
+    if response.content_type != MIMETYPE_APPLICATION_JSON:
+        raise HTTPInternalServerError(
+            reason=f"Profiling middleware is not compatible with {response.content_type=}",
```

Comment:
q for my learning: what does the `=` do here?

Reply:
When putting `=` after a variable the variables name will be added in the string. So if `myvar=42` then `f"{myvar=}"` will resolve to `"myvar=42"`